client/core/simnet_trade.go

```diff
@@ -461,93 +470,55 @@ func testMakerGhostingAfterTakerRedeem(s *simulationTest) error {
 	var qty, rate uint64 = 1 * s.lotSize, 250 * s.rateStep
 	s.client1.isSeller, s.client2.isSeller = true, false
 
-	c1OrderID, c2OrderID, err := s.placeTestOrders(qty, rate)
-	if err != nil {
-		return err
-	}
-
-	// Monitor trades and stop at order.TakerSwapCast
-	monitorTrades, ctx := errgroup.WithContext(context.Background())
-	monitorTrades.Go(func() error {
-		return s.monitorOrderMatchingAndTradeNeg(ctx, s.client1, c1OrderID, order.TakerSwapCast)
-	})
-	monitorTrades.Go(func() error {
-		return s.monitorOrderMatchingAndTradeNeg(ctx, s.client2, c2OrderID, order.TakerSwapCast)
-	})
-	if err = monitorTrades.Wait(); err != nil {
-		return err
-	}
-
-	// Resume trades but disable Maker's ability to notify the server
-	// after redeeming Taker's swap.
-	resumeTrade := func(ctx context.Context, client *simulationClient, orderID string) error {
-		tracker, err := client.findOrder(orderID)
-		if err != nil {
-			return err
-		}
-		finalStatus := order.MatchComplete
-		var disconnectClient bool
-		tracker.mtx.Lock()
-		for _, match := range tracker.matches {
-			side, status := match.Side, match.Status
-			client.log.Infof("trade %s paused at %s", token(match.MatchID[:]), status)
-			if side == order.Maker {
-				// Disconnecting the client will lock the
-				// tracker mutex, and so cannot be done with it already locked.
-				disconnectClient = true
-			} else {
-				client.log.Infof("%s: resuming trade negotiations to audit Maker's redeem", side)
-				client.noRedeemWait = true
+	c1FakeConns, c2FakeConns := s.client1.fakeConns, s.client2.fakeConns
+	c1FakeConns.close()
+	defer c1FakeConns.open()
+	c2FakeConns.close()
+	defer c2FakeConns.open()
+	anErr := errors.New("intentional error from test")
+	done := make(chan struct{})
+	wait := make(chan struct{})
+	go func() {
+		// Prevent the first redeemer, who must be maker, from sending
+		// redeem info to the server.
+		var bits uint8
+		for {
+			select {
+			case <-done:
+				close(wait)
+				return
+			case route := <-c1FakeConns.routeCh:
+				if route == msgjson.RedeemRoute {
+					if bits == 0 {
+						bits |= 0x1
+					}
+					if bits&0x1 != 0 {
+						c1FakeConns.errCh <- anErr
+						continue
+					}
+				}
+				c1FakeConns.errCh <- nil
+			case route := <-c2FakeConns.routeCh:
+				if route == msgjson.RedeemRoute {
+					if bits == 0 {
+						bits |= 0x01
+					}
+					if bits&0x01 != 0 {
```

Comment:
I'm not quite getting the intent of the bits.  You have `0x1` in the c1 case, and `0x01` in the c2 case, but those are the same number (same bit).  Did you want it to be `0x2` down here.  Is it needed to distinguish c1 from c2 in any test?  Could this be done with one (or two) bools?

Reply:
Oh, shoot. I meant to be using the `0b` prefix. And it would progress 0b1 -> 0b10 -> 0b100 but I'll change to what buck suggests since I was also confusing it.